src/streaming/ManifestLoader.js

```diff
@@ -42,25 +42,30 @@ import Errors from '../core/errors/Errors.js';
 import FactoryMaker from '../core/FactoryMaker.js';
 import DashParser from '../dash/parser/DashParser.js';
 
-function ManifestLoader(config) {
-
-    config = config || {};
-    const context = this.context;
-    const debug = config.debug;
-    const settings = config.settings;
-    const eventBus = EventBus(context).getInstance();
-    const urlUtils = URLUtils(context).getInstance();
-
+function ManifestLoader() {
```

Comment:
Why this change is needed? It is tested that anything else is still working?

Reply:
This was implemented because the manifestLoader was changed to a singleton. In MediaPlayer, the ListMpdController is initialized before the manifestLoader, and an instance of it is needed.

It shouldn't be an issue that it's a singleton since there isn't another place where a new instance of manifestLoader is created. 
Notice that the config code is moved to setConfig.

Anyway, after a quick look at the code, I think it might be possible to change the initialization order so that ListMpdController is initialized after manifestLoader (along with all playbackControllers), avoiding this change. I have to take a deeper look into it.